src/renderers/shared/fiber/ReactFiberBeginWork.js

```diff
@@ -683,6 +712,14 @@ module.exports = function<T, P, I, TI, PI, C, CX>(
     // Unmount the current children as if the component rendered null
     const nextChildren = null;
     reconcileChildren(current, workInProgress, nextChildren);
+
+    if (workInProgress.tag === ClassComponent) {
+      const instance = workInProgress.stateNode;
+      workInProgress.memoizedProps = instance.props;
+      workInProgress.memoizedState = instance.state;
```

Comment:
A bunch of tests fail because of this invariant: https://github.com/facebook/react/blob/24b8ba7340f5ce66df0b858ead480ca7103a8153/src/renderers/shared/fiber/ReactFiberClassComponent.js#L329